### DISCUSSION TITLE
Question about file archiver dependencies

### DISCUSSION THREAD

**Discussion by isf63** [tsujan/Arqiver#109] [Q&A] [answered]
I am running Arch Linux and have been running the arqiver-git package.

Looking at the dependencies, they are: `libarchive  gzip  p7zip  qt5-base  qt5-x11extras  qt5-svg`.

I've been wondering why arqiver depends on gzip and p7zip, when looking at https://github.com/libarchive/libarchive, libarchive _should_ handle both extracting and creating .gz and .7z archive formats.

Thanks!

**Comment by tsujan** [comment, marked as answer]
> libarchive should handle both extracting and creating .gz and .7z archive formats.

It's not just about 7z archives. There are several archives that `bsdtar` can't handle (correctly), while `7z` can, and conversely. So, both are needed for a good coverage.

Also, it makes no difference to Arqiver whether `p7zip` is installed ot `7-zip-full`. The former covers more archives (and I use it), while the latter handles some non-Linux archives better (e.g., dmg).

**Comment by tsujan** [comment]
A list of archive formats, for which `bsdtar` can't be trusted (found by trial and error):

https://github.com/tsujan/Arqiver/blob/e3002e0cfa3e124fbfebb8473d48bdbc4c911ba7/backends.cpp#L162-L172
